### PR TITLE
GUI ToolBox

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/github/weichware10/toolbox/Main.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import javafx.stage.WindowEvent;
+
+/**
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/Main.java:17]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.

File: src/main/java/github/weichware10/toolbox/Main.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ */
+public class Main extends Application {
+    public static void main(String[] args) {
+        launch(args);
+    }
+
+    @Override
+    public void start(Stage primaryStage) throws Exception {
+        primaryStage.setTitle("ToolBox");
+
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/Main.java:29]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line has trailing whitespaces.

File: src/main/java/github/weichware10/toolbox/Main.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+public class Main extends Application {
+    public static void main(String[] args) {
+        launch(args);
+    }
+
+    @Override
+    public void start(Stage primaryStage) throws Exception {
+        primaryStage.setTitle("ToolBox");
+        
+        Label trialIdExplenation = new Label("Insert the ID given in the Email");
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/Main.java:30]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Distance between variable 'trialIdExplenation' declaration and its first usage is 4, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

File: src/main/java/github/weichware10/toolbox/Main.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+        Button startTestButton = new Button("Start");
+        startTestButton.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent event) {
+                System.out.println("Test started");
+            }
+        });
+
+        VBox LayoutTestIdWindow = new VBox(10);
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/Main.java:43]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.LocalVariableNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Local variable name 'LayoutTestIdWindow' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$' (camelCase, start with lowercase, only alphanumerical).

File: src/main/java/github/weichware10/toolbox/Main.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        LayoutTestIdWindow.getChildren().addAll(trialIdExplenation, trialIdInput, startTestButton);
+        LayoutTestIdWindow.setAlignment(Pos.CENTER);
+
+        Scene toolBoxHome = new Scene(LayoutTestIdWindow, 400, 400);
+
+        //----------end----------
+        //set the action after the window got closed
+        primaryStage.setOnCloseRequest(new EventHandler<WindowEvent>() {
+            @Override
+            public void handle(WindowEvent event){
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/Main.java:54]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '{' is not preceded with whitespace.

File: src/main/java/github/weichware10/toolbox/Main.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                event.consume();
+                //Close window with yes or no
+                closeProgramm(primaryStage);
+            }
+        });
+        primaryStage.setScene(toolBoxHome);
+        primaryStage.show();
+    }
+
+    private void closeProgramm(Stage window){
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/Main.java:65]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '{' is not preceded with whitespace.

File: src/main/java/github/weichware10/toolbox/Main.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        boolean answer = ConfirmBox.display("End the programm", "Sure you want to quit ?");
+        if(answer){
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/Main.java:67]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/Main.java:67]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)

Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/Main.java:67]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '{' is not preceded with whitespace.

File: src/main/java/github/weichware10/toolbox/ConfirmBox.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.layout.VBox;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+
+//How to Communicate between windows
+public class ConfirmBox {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/ConfirmBox.java:14]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocTypeCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/github/weichware10/toolbox/ConfirmBox.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    static boolean answer;
+
+    public static boolean display(String title, String message){
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/ConfirmBox.java:18]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '{' is not preceded with whitespace.

File: src/main/java/github/weichware10/toolbox/ConfirmBox.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+        final Stage window = new Stage();
+
+        window.initModality(Modality.APPLICATION_MODAL);
+        window.setTitle(title);
+        window.setMinWidth(250);
+
+        Label label1 = new Label();
+        label1.setText(message);
+
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/ConfirmBox.java:27]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line has trailing whitespaces.

File: src/main/java/github/weichware10/toolbox/ConfirmBox.java [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+        window.setMinWidth(250);
+
+        Label label1 = new Label();
+        label1.setText(message);
+        
+        Button yesButton = new Button("Yes");
+        Button noButton = new Button("No");
+        yesButton.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent event){
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/ConfirmBox.java:32]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '{' is not preceded with whitespace.

File: src/main/java/github/weichware10/toolbox/ConfirmBox.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+                answer = true;
+                window.close();
+            }
+        });
+        noButton.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent event){
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/ConfirmBox.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '{' is not preceded with whitespace.

File: src/main/java/github/weichware10/toolbox/ConfirmBox.java [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+                answer = false;
+                window.close();
+            }
+        });
+
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/ConfirmBox.java:44]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line has trailing whitespaces.

File: src/main/java/github/weichware10/toolbox/ConfirmBox.java [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+        layout.getChildren().addAll(label1, yesButton, noButton);
+        layout.setAlignment(Pos.CENTER);
+
+        Scene scene = new Scene(layout);
+        window.setScene(scene);
+        window.showAndWait();
+
+        return answer;
+    }
+
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/ConfirmBox.java:56]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line has trailing whitespaces.

File: src/main/java/github/weichware10/toolbox/zoommaps/Zoombildschirm.java [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+/**
+ * Kümmert sich um das Design des Tests Zoommaps.
+ */
+public class ZoomBildschirm {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/zoommaps/Zoombildschirm.java:19]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.OuterTypeFilenameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>The name of the outer type and the file do not match.

**Comment by Rotwolf** [comment]
Das wurde jetzt also auf den nächsten Meilenstein verschoben?

File: src/main/java/github/weichware10/toolbox/eyetracking/EyeTrackingBildschirm.java [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+/**
+ * Kümmert sich um das Design des Tests Zoommaps.
+ */
+public class EyeTrackingBildschirm {
```
Comment (grenoui11e) [inline src/main/java/github/weichware10/toolbox/eyetracking/EyeTrackingBildschirm.java:19]: muss das Eyetracking heißen wenn dass Zoommaps ist?

File: src/main/java/github/weichware10/toolbox/eyetracking/EyeTrackingBildschirm.java [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+/**
+ * Kümmert sich um das Design des Tests Zoommaps.
```
Comment (grenoui11e) [inline src/main/java/github/weichware10/toolbox/eyetracking/EyeTrackingBildschirm.java:17]:
```suggestion
 * Kümmert sich um das Design des Tests Eyetracking.
```

**Review by grenoui11e — DISMISSED** [review]
In EyetrackingBildschirm.java ist ein Schreibfehler

File: src/main/java/github/weichware10/toolbox/codecharts/CodeChartBildschirm.java [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+
+        Button startButton = new Button("Start");
+        startButton.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent event) {
+                Endbildschirm.display(primaryStage, configClient);
+            }
+        });
+
+        Button endButton = new Button("Abschließen");
```
Comment (Rotwolf) [inline src/main/java/github/weichware10/toolbox/codecharts/CodeChartBildschirm.java:38]: Kann man da nicht auf Abschließen drücken ohne was gemacht zu haben?
Kommt bestimmt später noch, oder?

Comment (joshuajeschek) [reply src/main/java/github/weichware10/toolbox/codecharts/CodeChartBildschirm.java:38]: Das ist glaube zurzeit ein Platzhalter oder @justinexner?

Comment (justinexner) [reply src/main/java/github/weichware10/toolbox/codecharts/CodeChartBildschirm.java:38]: Richtig, ich brauche ja erst die Bedingung, dass irgendwas gemacht wurde und dafür brauche ich den Code wodurch der Versuch läuft, aber danke für den Hinweis, kann ich als Kommentar hinzufügen 🤝

File: src/main/java/github/weichware10/toolbox/gui/Startbildschirm.java [hunk 19]
```diff
@@ -0,0 +1,280 @@
...
+        BorderPane layoutTestIdWindow = new BorderPane();
+        layoutTestIdWindow.setTop(testMenuBar);
+        layoutTestIdWindow.setCenter(layoutTestIdCenterWindow);
+
+        toolBoxHome = new Scene(layoutTestIdWindow, 400, 400);
+
+        primaryStage.setScene(toolBoxHome);
+    }
+
+    /**
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Startbildschirm.java:256]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>First sentence of Javadoc is missing an ending period.

File: src/main/java/github/weichware10/toolbox/gui/Startbildschirm.java [hunk 20]
```diff
@@ -0,0 +1,280 @@
...
+     * Setzt die Datenbankverbindung auf die Werte in der env Datei
+     * @return Erfolgsboolean
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Startbildschirm.java:258]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@return' should be preceded with an empty line.

**Review by Rotwolf — APPROVED** [review]
Passt

File: src/main/java/github/weichware10/toolbox/gui/Startbildschirm.java [hunk 21]
```diff
@@ -0,0 +1,285 @@
...
+import javafx.scene.control.TextField;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.ColumnConstraints;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+import javafx.scene.text.Text;
+import javafx.stage.FileChooser;
+import javafx.stage.Screen;
+import javafx.stage.FileChooser.ExtensionFilter;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/toolbox/gui/Startbildschirm.java:37]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.stage.FileChooser.ExtensionFilter' import. Should be before 'javafx.stage.Screen'.

**Review by ghost — APPROVED** [review]
Sehr gut gemacht

**Pull request merged**